src/stories/InlineAutocomplete.stories.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {useState} from 'react'
+import {Meta} from '@storybook/react'
+
+import {BaseStyles, Box, Textarea, ThemeProvider, ActionList, Avatar, ActionListItemProps, FormControl} from '..'
+import InlineAutocomplete, {ShowSuggestionsEvent, Suggestions} from '../InlineAutocomplete'
+
+export default {
+  title: 'Forms/InlineAutocomplete',
+  component: InlineAutocomplete,
+  decorators: [
+    Story => {
+      return (
+        <ThemeProvider>
+          <BaseStyles>
+            <Box paddingTop={5}>{Story()}</Box>
+          </BaseStyles>
+        </ThemeProvider>
+      )
+    }
+  ],
+  argTypes: {}
+} as Meta
+
+interface User {
+  login: string
+  name: string
+  avatar: string
+  type: 'user' | 'organization'
+}
+
+const sampleUsers: User[] = [
+  {login: 'monalisa', name: 'Monalisa Octocat', avatar: 'https://avatars.githubusercontent.com/github', type: 'user'},
+  {login: 'primer', name: 'Primer', avatar: 'https://avatars.githubusercontent.com/primer', type: 'organization'},
+  {login: 'github', name: 'GitHub', avatar: 'https://avatars.githubusercontent.com/github', type: 'organization'}
+]
+
+const UserSuggestion = ({user, ...props}: {user: User} & ActionListItemProps) => (
+  <ActionList.Item {...props}>
+    <ActionList.LeadingVisual>
+      <Avatar src={user.avatar} square={user.type === 'organization'} />
+    </ActionList.LeadingVisual>
+    {user.name}
+    <ActionList.Description>{user.login}</ActionList.Description>
+  </ActionList.Item>
+)
+
+export const Default = () => {
+  const [suggestions, setSuggestions] = useState<Suggestions | null>(null)
+
+  const onShowSuggestions = (event: ShowSuggestionsEvent) => {
+    const filteredUsers = sampleUsers.filter(
```

Comment:
In the other Autocomplete component we default to an alphanumeric filter, but allow that to be overridden using the `filterFn` prop. https://github.com/primer/react/blob/c7bbd06fd7c67690ed7a612737f79662444d388b/src/Autocomplete/AutocompleteMenu.tsx#L206

Reply:
Personally, I think this component is a lot easier to use if it doesn't manage its own state - in other words, the parent component handles filtering and then tells this component what to display.

That allows a lot of flexibility without a ton of props - the parent component can use a custom filtering function, control the sorting (very important when using fuzzy filtering), control the number of visible suggestions, and manage multiple sets of suggestions (when there are multiple types of triggers).

If we want to keep it consistent, I think I would eventually prefer to move `Autocomplete` in this stateless direction as well.

Reply:
You make a great point. Go ahead and keep the filtering as-is.

We could reconsider changing the Autocomplete API in the next breaking change.